src/rail_gui.cpp

```diff
@@ -243,13 +248,13 @@ static void GenericPlaceSignals(TileIndex tile)
 			SB(p1, 4, 1, _cur_signal_variant);
 			SB(p1, 5, 3, _cur_signal_type);
 			SB(p1, 8, 1, _convert_signal_button);
-			SB(p1, 9, 6, cycle_bounds[_settings_client.gui.cycle_signal_types]);
+			SB(p1, 9, 6, ((_settings_client.gui.cycle_signal_types == SIGNAL_CYCLE_ALL && _settings_client.gui.signal_gui_mode == SIGNAL_GUI_ALL) ? (SIGTYPE_NORMAL | (SIGTYPE_LAST << 3)) : (SIGTYPE_PBS | (SIGTYPE_LAST << 3)) ));
```

Comment:
Maybe put this in a temporary variable (the `(_settings_client.gui.cycle_signal_types == SIGNAL_CYCLE_ALL && _settings_client.gui.signal_gui_mode == SIGNAL_GUI_ALL)` part) to increase readability? As it is now duplicated twice, and it is a pretty long statement.